simd/src/Kokkos_SIMD_Common_Math.hpp

```diff
@@ -0,0 +1,262 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_SIMD_COMMON_MATH_HPP
+#define KOKKOS_SIMD_COMMON_MATH_HPP
+
+#include <Kokkos_Core.hpp>  // Kokkos::min, etc.
+
+namespace Kokkos {
+
+namespace Experimental {
+
+template <class T, class Abi>
+class simd;
+
+template <class T, class Abi>
+class simd_mask;
+
+template <class M, class T>
+class const_where_expression;
```

Comment:
Why did you go for forward declarations rather than including `<Kokkos_SIMD_Common.hpp>`?

Reply:
To minimize dependencies (even though it probably doesn't matter much here). We should be able to use this header just using the `SIMD` specializations and not the fallback. 
In the end, this pull request shows that the order of overload declarations matters, and including `Kokkos_SIMD_Common.hpp` might suggest that it would be fine to include that file before the overloads for the `simd` specializations.